manifests/plugin.pp

```diff
@@ -22,16 +22,24 @@
   $config_content  = undef,
   $update_url      = undef,
   $plugin_dir      = '/var/lib/jenkins/plugins',
-  $username        = 'jenkins',
-  $group           = 'jenkins',
   $enabled         = true,
   $create_user     = true,
   $source          = undef,
 ) {
   include ::jenkins::params
 
-  $plugin            = "${name}.hpi"
-  $plugin_parent_dir = inline_template('<%= @plugin_dir.split(\'/\')[0..-2].join(\'/\') %>')
+  $plugin     = "${name}.hpi"
+  $homedir    = $::jenkins::params::homedir
+  $jenkinsdir = $::jenkins::params::jenkinsdir
+  $username   = $::jenkins::params::jenkins_user
+  $group      = $::jenkins::params::jenkins_group
+
+  if $plugin_dir == '/var/lib/jenkins/plugins' {
+    $real_plugin_dir = $::jenkins::params::default_plugin_dir
+  } else {
+    $real_plugin_dir = $plugin_dir
+  }
+
```

Comment:
changed to the suggested selector.
The idea behind is that, the default value for the parameter plugin_dir was originally hard-coded to '/var/lib/jenkins/plugins', and I did not wanted to break existing configurations that relied on it.
Since this is a defined type, and I remember I had trouble with just assigning 
$plugin_dir = $::jenkins::params::default_plugin_dir as the parameter to the defined type, even if the include ::jenkins::params is there in the list of parameters to the defined type. Therefore this kind of workaround.